src/zenml/cli/server.py

```diff
@@ -785,7 +786,15 @@ def connect(
     assert store_config_class is not None
 
     store_config = store_config_class.parse_obj(store_dict)
-    GlobalConfiguration().set_store(store_config)
+    try:
+        GlobalConfiguration().set_store(store_config)
+    except IllegalOperationError:
+        cli_utils.warning(
+            f"User '{username}' has no permissions attached to its role with "
+            f"which to access the server at '{url}'. Please ask the server "
+            f"administrator to assign a role with permissions to your "
+            f"username."
+        )
```

Comment:
```suggestion
    except IllegalOperationError as e:
        cli_utils.warning(
            f"User '{username}' does not have sufficient permissions to "
            f"to access the server at '{url}'. Please ask the server "
            f"administrator to assign a role with permissions to your "
            f"username: {str(e)}"
        )
```

Or something similar. The reason being that the IllegalOperationError can be thrown in cases where the user only has read permissions or doesn't have permissions to modify the current project, or other situations such as these, so the message should be more generic and also include the original error.